docs/components/index.md

```diff
@@ -0,0 +1,360 @@
+
+*BlazorRocks.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRocks.cshtml?start=1&end=8)]
+
+*BlazorRocksBase.cs*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRocksBase.cs)]
+
+The base class must derive from `BlazorComponent` in the `Microsoft.AspNetCore.Blazor.Components` namespace.
+
+### Render C# expressions
+
+C# expressions are rendered by prefixing the `@` symbol. For example, a C# field is rendered by prefixing `@` to the field name. The following example evaluates and renders:
+
+* The `_HeadingFontStyle` field to the CSS property value for `font-style`.
+* `_HeadingText` to the content of the `<h1>` element.
+
+```cshtml
+<h1 style="font-style:@_HeadingFontStyle">@_HeadingText</h1>
+```
+
+## Two-way databinding
+
+Two-way databinding to both components and DOM elements is accomplished with the `bind` attribute. The following example binds the `ItalicsCheck` property to the check box's checked state:
+
+```cshtml
+<input type="checkbox" class="form-check-input" 
+    id="italicsCheck" bind="@ItalicsCheck">
+```
+
+When the check box is selected and cleared, the property's value is updated to `true` and `false`, respectively.
+
+The check box is updated in the UI only when the component is rendered, not in response to changing the property's value. Since components render themselves after event handler code executes, property updates are usually reflected in the UI immediately.
+
+Using `bind` with a `CurrentValue` (`<input bind="@CurrentValue" />`) is essentially equilivant to the following:
+
+```cshtml
+<input value="@CurrentValue" onchange="@((UIValueEventArgs __e) => CurrentValue = __e.Value) />
+```
+
+When the component is rendered, the `value` of the input element comes from the `CurrentValue` property. When the user types in the text box, the `onchange` is fired and the `CurrentValue` property is set to the changed value. In reality, the code generation is a little more complex because `bind` deals with a few cases where type conversions are performed. In principle, `bind` associates the current value of an expression with a `value` attribute and handles changes using the registered handler.
+
+**Format strings**
+
+Two-way databinding works with [DateTime](https://docs.microsoft.com/dotnet/api/system.datetime) format strings (but not other format expressions at this time, such as currency or number formats):
+
+```cshtml
+<input bind="@StartDate" format-value="yyyy-MM-dd" />
+
+@functions {
+    public DateTime StartDate { get; set; } = 
+        new DateTime(2020, 1, 1);
+}
+```
+
+**Conditional attributes**
+
+Blazor conditionally renders attributes based on the bound .NET value. If the value bound is `false` or `null`, Blazor won't render the attribute. If the value is `true`, the attribute is rendered minimized.
```

Comment:
There's no binding here, so let's not use the term "bound". We just conditionally render attributes according to the specified value.